TopicN1/src/CashStrategy.java

```diff
@@ -0,0 +1,24 @@
+import java.util.Calendar;
+
+public class CashStrategy implements PaymentStrategy {
+
+	private int counter;
+	private MailList mailList;
+    private Calendar calendar;
+
+    public CashStrategy(MailList mailList){
+    	this.mailList=mailList;
```

Comment:
MailList reference should be only in shoopingCart class instead in each of payment class in order to apply correctly Observer pattern.